pkg/prometheus/promcfg.go

```diff
@@ -937,6 +937,10 @@ func (cg *ConfigGenerator) generateProbeConfig(
 	switch {
 	case m.Spec.Targets.StaticConfig != nil:
 		// Generate static_config section.
+		// Hosts takes precedence over Targets, since the latter is deprecated.
+		if m.Spec.Targets.StaticConfig.Hosts != nil {
+			m.Spec.Targets.StaticConfig.Targets = m.Spec.Targets.StaticConfig.Hosts
+		}
```

Comment:
Not sure how we handled similar deprecations in the past (if we did at all), but any reason why not doing something like below?
```suggestion
			m.Spec.Targets.StaticConfig.Targets = append(m.Spec.Targets.StaticConfig.Targets, ...m.Spec.Targets.StaticConfig.Hosts)
```

Reply:
Since we are encouraging folks to use the newer `hosts` over the older `static` in this patch, I thought it'd make sense for the former to take precedence when both are specified. I pointed this out in the change log above as well to make it explicit for the users, but LMK if that doesn't make sense, and I'll make the change. :)

Reply:
We do something similar with `port` and `targetPort` for the endpoints:

https://github.com/prometheus-operator/prometheus-operator/blob/63e15b7eb4897653d2462f8f41826743393a1f89/pkg/prometheus/promcfg.go#L794-L803